src/extra/stream/client.ts

```diff
@@ -83,6 +86,84 @@ export class StreamClient extends EventDispatcher implements Stream {
     }
   }
 
+  /**
+   * Get the available livestream resolutions for a camera component. If the
+   * stream client cannot find any available resolutions, an empty list will be
+   * returned.
+   *
+   * @param resourceName - The name of a camera component.
+   * @returns A list of available resolutions for livestreaming.
+   */
+  async getOptions(resourceName: string): Promise<Resolution[]> {
+    const fetchOptions = async (name: string): Promise<Resolution[]> => {
+      const request = new GetStreamOptionsRequest({ name });
+      this.options.requestLogger?.(request);
+      try {
+        const response = await this.client.getStreamOptions(request);
+        return response.resolutions;
+      } catch {
+        return [];
+      }
+    };
+
+    const trackName = getValidSDPTrackName(resourceName);
+    let resolutions = await fetchOptions(trackName);
+    if (resolutions.length > 0) {
+      return resolutions;
+    }
+    // Second attempt with resource name
+    resolutions = await fetchOptions(resourceName);
+    return resolutions;
+  }
+
+  /**
+   * Set the livestream options for a camera component. This will change the
+   * resolution of the stream to the specified width and height.
+   *
+   * @param name - The name of a camera component.
+   * @param width - The width of the resolution.
+   * @param height - The height of the resolution.
+   */
+  async setOptions(name: string, width: number, height: number) {
+    const request = new SetStreamOptionsRequest({
```

Comment:
do we need any error checking that the width/height parameters are set within a specific bound?

Reply:
All of the GRPC field checks should happen in the server endpoint and it should hopefully error out here with the correct message. Would it be good to add additional checks here in the client before sending?

Reply:
ah got it. if the checks are happening elsewhere then we def don't need to duplicate it